src/dependency.cr

```diff
@@ -12,18 +12,23 @@ module Shards
       end
     end
 
-    def initialize(@name)
```

Comment:
Note that `protected` is the default visibility for `initialize`, so this is a bit redundant.

Reply:
And the visibility isn't reflected on the `new` constructor?

Reply:
No... but it's a good point, now that you mention it. What works is using `private`, but then you can't access it from within the same namespace. Hmm...